Sources/ScreenObject/ScreenObject.swift

```diff
@@ -10,8 +10,10 @@ open class ScreenObject {
     /// initialization time.
     public let app: XCUIApplication
 
+    private let expectedElementGetter: (XCUIApplication) -> XCUIElement
+
     /// The `XCUIElement` used to evaluate whether the screen is loaded at runtime.
-    public let expectedElement: XCUIElement
+    public var expectedElement: XCUIElement { expectedElementGetter(app) }
```

Comment:
Basically, instead of running the closure that takes a `XCUIApplication` as input an returns the expected `XCUIElement` only when the screen is created, we store the closure itself and run it every time we want to access the expected element.

Reply:
I think this approach is much more robust 👍